packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -838,7 +952,12 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
               FACEBOOK_JNI_THROW_PENDING_EXCEPTION();
             } catch (...) {
               TMPL::asyncMethodCallExecutionFail(moduleName, methodName, id);
-              throw;
+              if (rejectTurboModulePromiseOnNativeError()) {
+                  auto exception = std::current_exception();
+                  rejectWithException(rejectCallback, exception, jsInvocationStack);
+              } else {
+                  throw;
+              }
```

Comment:
Just for consistency, nil out rejectCallback after calling it.

```suggestion
              if (rejectTurboModulePromiseOnNativeError() && rejectCallback) {
                  auto exception = std::current_exception();
                  rejectWithException(*rejectCallback, exception, jsInvocationStack);
                  rejectCallback = std::nullopt;
              } else {
                  throw;
              }
```